frankenphp.go

```diff
@@ -501,6 +501,80 @@ func ServeHTTP(responseWriter http.ResponseWriter, request *http.Request) error
 	return nil
 }
 
+//export go_putenv
+func go_putenv(str *C.char, length C.int) C.bool {
+	// Create a byte slice from C string with a specified length
+	s := C.GoBytes(unsafe.Pointer(str), length)
+
+	// Convert byte slice to string
+	envString := string(s)
+
+	// Check if '=' is present in the string
+	if idx := strings.IndexByte(envString, '='); idx != -1 && idx < len(envString)-1 {
+		// '=' found, set the environment variable
+		key := envString[:idx]
+		val := envString[idx+1:]
+
+		if os.Setenv(key, val) != nil {
+			return false // Failure
+		}
+	} else {
+		// No '=', unset the environment variable
+		if os.Unsetenv(envString) != nil {
+			return false // Failure
+		}
+	}
+
+	return true // Success
+}
+
+//export go_getfullenv
+func go_getfullenv(threadIndex C.uintptr_t) (*C.go_string, C.size_t) {
+	thread := phpThreads[threadIndex]
+
+	env := os.Environ()
+	goStrings := make([]C.go_string, len(env)*2)
+
+	for i, envVar := range env {
+		key, val, _ := strings.Cut(envVar, "=")
+		k := unsafe.StringData(key)
+		v := unsafe.StringData(val)
+		thread.Pin(k)
+		thread.Pin(v)
```

Comment:
I'm on a phone so I may miss something but it looks like the variable is never unpin.

Reply:
Since the only option is to unpin _everything_ and I don't know what is currently pinned/being used, I had assumed there was a single unpin at the end of the request.

It seems this is not the case, but it will be unpinned at the beginning of the next request.

Reply:
Indeed... Maybe could we unpin at the end of the request to allow Go to collect garbage eagerly?

Reply:
done